### DISCUSSION TITLE
Workshop-Relay seems to be losing 10 monopoly Algos on a local devnet. Any idea where I should look

### DISCUSSION THREAD

**Discussion by JamesLampert** [reach-sh/reach-lang#832] [Q&A] [answered]
I've just tested the "Relay" workshop exercise, and my results don't seem to quite match what the workshop page says they should look like:

Where the last two lines of the results shown on the workshop page are:
```
Alice went from 100.0 to 74.999999999999804065.
Bob went from 100.0 to 123.999999999999979.
```

what I get is:
```
Alice went from 100 to 74.996.
Bob went from 100 to 114.999.
```

So where should I start looking for the missing 10 monopoly Algos?

I'm using the canned front-end, adjusting only to match the names to my backend, and near as I can tell, my backend is equivalent to the canned one.

**Comment by jeapostrophe** [comment, marked as answer]
That's the expected result. You should read the frontend code more closely --- https://github.com/reach-sh/reach-lang/blob/master/examples/workshop-relay/index.mjs#L36-L37

**Reply by jeapostrophe** [reply]
@TheChronicMonster Will you please generate new example output for the Web page?

**Reply by JamesLampert** [reply]
Hmm. Fascinating.

**Reply by JamesLampert** [reply]
Back in April, I noted (incorrectly) that changing the statement from
`transfer(amount).to(bob);`
to
`transfer(balance()).to(bob);`

Today, I found that made no difference at all. I also found that if I added statements to capture and display the relay balance, I got:
`Relay went from 0.1 to 10.1.`

So how _does_ Bob get his 10 algos back?

**Reply by JamesLampert** [reply]
And for that matter, I don't see any funds actually passing through the relay account itself. In both my solution and the solution on the workshop page, they appear to simply be passed directly from Alice into the contract, and from the contract directly to Bob.

**Reply by jeapostrophe** [reply]
The code that I linked shows Bob giving 10 ALGOs to the relay. These cannot be taken back by any means. You are correct about the path of the 25 ALGOs that the workshop is about.
